plugins/modules/cloud/misc/proxmox_nic.py

```diff
@@ -142,11 +142,18 @@
     HAS_PROXMOXER = False
 
 from ansible.module_utils.basic import AnsibleModule, env_fallback
-from ansible.module_utils._text import to_native
 
 
-def get_vmid(proxmox, name):
-    return [vm['vmid'] for vm in proxmox.cluster.resources.get(type='vm') if vm.get('name') == name]
+def get_vmid(module, proxmox, name):
+    try:
+        vms = [vm['vmid'] for vm in proxmox.cluster.resources.get(type='vm') if vm.get('name') == name]
+    except Exception:
+        module.fail_json(msg='VM with name = %s does not exist in cluster' % name)
```

Comment:
Moving the try block here means that the exception would be specifically related to a bad response from the PVE API and not necessarily that the VM doesn't exist. So maybe reporting the exception message would make sense here.

Reply:
Thanks, added in the latest commit.